lib/curl_sasl_sspi.c
=======================================================
lhs: 100644 | 422b2f64c1d432248dc90195de00c9e314b3c579
rhs: 100644 | df4da96456d9019678280378214b366488f9cb67
---@@ -416,8 +416,11 @@ CURLcode Curl_sasl_create_gssapi_user_message(struct SessionHandle *data,
                                                &resp_desc, &attrs,
                                                &tsDummy);
 
-  if(status != SEC_E_OK && status != SEC_I_CONTINUE_NEEDED)
+  if(status != SEC_E_OK && status != SEC_I_CONTINUE_NEEDED) {
+    Curl_safefree(chlg);
+
     return CURLE_RECV_ERROR;
+  }
 
   if(memcmp(&context, krb5->context, sizeof(context))) {
     s_pSecFn->DeleteSecurityContext(krb5->context);
@@ -431,6 +434,9 @@ CURLcode Curl_sasl_create_gssapi_user_message(struct SessionHandle *data,
                                 resp_buf.cbBuffer, outptr, outlen);
   }
 
+  /* Free the decoded challenge */
+  Curl_safefree(chlg);
+
   return result;
 }
 

---